database/ldb/leveldb.go

```diff
@@ -143,8 +143,13 @@ blocknarrow:
 
 	// Load the last block whose transactions have been indexed by address.
 	if sha, idx, err := ldb.fetchAddrIndexTip(); err == nil {
-		ldb.lastAddrIndexBlkSha = *sha
-		ldb.lastAddrIndexBlkIdx = idx
+		if err = ldb.checkAddrIndexVersion(); err == nil {
+			ldb.lastAddrIndexBlkSha = *sha
+			ldb.lastAddrIndexBlkIdx = idx
+		} else {
+			ldb.deleteOldAddrIndex()
+			ldb.DeleteAddrIndex()
```

Comment:
I think it'd be nice to throw a logging statement in here detailing the detection and deletion of the deprecated `addrindex` version. Otherwise a node operator might be a bit confused if `btcd` comes up and hangs for a bit while deleting the old index without any informational output.